### ISSUE TITLE
[ suggestions ] enabled_plugins is not installed by default (config.toml) and completions

### ISSUE THREAD

**Issue by micisse** [triyanox/lla#62]
Hello, as part of an _automated installation_ and in the case of “**lla**”, I use the configuration file with fields that can be dynamic. For example, when I set the default list to

```toml
...
default_sort = "{{ var_dynamic_field_for_example }}"          # Possible values: "name", "size", "date"
enabled_plugins = ["git_status", "file_hash", "sizeviz"]  # List of enabled plugins
...
```
**All specified plugins are not installed by default, the specified Plugins folder is empty and I have received a warning**. 
![Capture d’écran_2024-12-16_14-58-22](https://github.com/user-attachments/assets/fa38896d-033a-4ca6-ad7b-ebbed21fe9bf)
![Capture d’écran_2024-12-16_15-13-47](https://github.com/user-attachments/assets/b8832229-9d7d-4802-a170-553221846950)

I have to install them manually. 

* Perhaps a suggestion would be to allow full installation of all plugins specified in the configuration with URLS (github or other) as part of a plugin external to “**lla**” or developed by others.

For example:
```toml
...
enabled_plugins = ["categorizer <https://path.to/categorizer>", "file_hash <url>", ...etc]  # List of enabled plugins
# or URLs directly, knowing that plugins have a basic structure
enabled_plugins = ["https://path.to/...", ...etc]  # List of enabled plugins
...
```
* Another suggestion that could also improve the tool is autocompletion. I saw at **eza**, for example, a “**[completions](https://github.com/eza-community/eza/tree/main/completions)**” folder for bash, zsh ...etc. This can be useful and should also be avoided `lla --help`
![Capture d’écran_2024-12-16_15-05-08](https://github.com/user-attachments/assets/d93c4344-1277-414c-b329-5edfbde69e53)

```bash
lla <tab to see options list or completions>
```

**Comment by triyanox** [comment]
@micisse Thank you for your excellent suggestions! Shell completions are planned for future releases and will support commonly used shells like nu, fish, bash, and zsh.

Regarding plugins, the minimal installation is intentional, as most users prefer to have full control over what they install on their systems. As documented, you can install plugins in several ways:

- Using git:` lla install --git https://github.com/triyanox/lla # you can use any repo that has plugins` you can install from any repository containing plugins (they will be built and placed in the correct location) and that also takes care of updating the plugins using `lla update` or `lla update plugin_name`
- Using the lla install --dir command to install from a local directory, which is particularly useful during development

**Comment by micisse** [comment]
@triyanox Noted for “Shells completions” 👍. However, wouldn't it be better to have an automatic installation as long as the plugins are specified in the configuration file? **For example, as I was saying, in the case of an automated installation from source using “ansible”, the installation will always be minimal because it's the user who composes his own configuration file and has total control**. This idea came to me from **RUST** for example. When you install it **from source**, it automatically downloads the specified items if they're not already installed. (plugins) Current methods are manual and sequenced. It was a nice little suggestion. **Do you want me to close the ticket or leave it for “Shell completions”?**

**Comment by triyanox** [comment]
I see I think the idea of the config based installation is really interesting for sure, we just need to just get it done right to make it straight forward and simple, I will consider it for the upcoming releases for sure, maybe something like how neovim handles extensions will be quite appealing for sure!

For completions just leave this issue open for now till the feature is implemented in the upcoming releases!

**Comment by triyanox** [comment]
Hey there @micisse the completions are shipped on `v0.3.6` and you can use `lla completion` or use the generated completions in /completions directory in the repo if you have a custom setup.

For the plugin installation enhancements will be a future release once I can find a nice way to implement it with good user experience.

Thanks again for your suggestions @micisse
